### ISSUE TITLE
Partial yield coverage

### ISSUE THREAD

**Issue by nedbat** [coveragepy/coveragepy#402]
*Originally reported by* **Anonymous**

----------------------------------------

Hi,
I think I discovered a bug. I opened a branch for it here:
https://github.com/coala-analyzer/coala/commits/Makman2/coverage-bug

If you clone coala (github.com/coala-analyzer/coala), switch to branch `Makman2/coverage-bug` and run coverage over it (via `./run_tests.py -cH`, `-c` is for coverage and `-H` display HTML report), in the file `SpaceConsistencyBear.py` I get partial coverage for a yield statement.

Changing the code here
```
            if len(result_texts) > 0:
                 diff = Diff()
                 diff.change_line(line_number + 1, line, replacement)
                 yield PatchResult(self,
                                  _("Line contains following spacing "
                                    "inconsistencies:") +
                                  ''.join("\n- " + string
                                          for string in result_texts),
                                   {filename: diff},
                                   filename,
                                   line_nr=line_number+1)
```
to
```
            if len(result_texts) > 0:
                 diff = Diff()
                 diff.change_line(line_number + 1, line, replacement)

                 inconsistencies = ''.join("\n- " + string
                                           for string in result_texts),

                 yield PatchResult(self,
                                  _("Line contains following spacing "
                                    "inconsistencies:") + inconsistencies,
                                   {filename: diff},
                                   filename,
                                   line_nr=line_number+1)
```
solves the problem, but I only moved out the generator expression inside the yield, which in fact is no code-flow change.

----------------------------------------
- Bitbucket: https://bitbucket.org/ned/coveragepy/issue/402

**Comment by nedbat** [comment]
I tried this with coverage.py 4.0 (beta 2), and didn't see the issue, so I think it is fixed, as a duplicate of either #308 or #324.

**Comment by nedbat** [comment]
Duplicate of #308.

**Comment by nedbat** [comment]
*Original comment by* **Mischa Krüger (Bitbucket: [Makman2](https://bitbucket.org/Makman2), GitHub: [Makman2](https://github.com/Makman2))**

----------------------------------------

Hi, I've found my lost account again :P
Thanks for looking at it, nice it is already solved in the 4.0 beta :)
